lib/views/git-tab-view.js

```diff
@@ -32,7 +34,30 @@ export default class GitTabView {
   }
 
   render() {
-    if (this.props.repository.showGitTabInit()) {
+    if (this.props.repository.isTooLarge()) {
+      const workingDir = this.props.repository.getWorkingDirectoryPath();
+      return (
+        <div className="github-Panel is-empty" tabIndex="-1">
+          <div ref="noRepoMessage" className="github-Panel no-repository">
+            <div className="large-icon">
+              <GitLogo />
+            </div>
+            {workingDir === os.homedir()
+              ? <div className="initialize-repo-description">
+                  The number of changed files in the repository at <strong>{workingDir}</strong> is
+                  too large for Atom to manage. In addition, the repository is located at your
+                  home directory, which may be unintended. <a href="https://gist.github.com/BinaryMuse/something">More details</a>
```

Comment:
The best error messages help the user understand how to resolve the error, and it would be difficult to explain resolution steps in sufficient detail here. I :heart: that you're linking to a resource that can provide more details. :zap:

In the scenario where we detect a repository in the user's home directory, am I correct in thinking that the external resource will recommend that the user consider deleting that repository (assuming they haven't intentionally created that repository)?

Reply:
> Might be helpful to have a title like "Too many changes". So that when you keep seeing these kind of messages, it's quick to know what they are about and you don't have to read through them every time. -- https://github.com/atom/github/pull/1153#issuecomment-327656181

I agree.

---

> The number of changed files in the repository at **/Users/mtilley/** is too large for Atom to manage.

If we reorder this a bit, we can shorten this sentence. I also think the reordering increases the simplicity of the sentence, which increases the readability:

> The repository at **/Users/mtilley/** has too many changed files to display in Atom.

---

Putting this all together, I think this does a nice job of conveying the situation:

> ### Too many changes
> 
> The repository at **/Users/mtilley/** has too many changed files to display in Atom. In addition, the repository is located at your home directory, which may be unintended. [More details](#)

Reply:
Thanks @jasonrudolph! Love what you came up with ✨ 

> In the scenario where we detect a repository in the user's home directory, am I correct in thinking that the external resource will recommend that the user consider deleting that repository (assuming they haven't intentionally created that repository)?

yup! 👍